runelite-client/src/main/java/net/runelite/client/plugins/barrows/BarrowsOverlay.java

```diff
@@ -0,0 +1,196 @@
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.barrows;
+
+import com.google.common.primitives.Ints;
+import net.runelite.api.Client;
+import net.runelite.api.GameObject;
+import net.runelite.api.NPC;
+import net.runelite.api.ObjectComposition;
+import net.runelite.api.Perspective;
+import net.runelite.api.Player;
+import net.runelite.api.Query;
+import net.runelite.api.WallObject;
+import net.runelite.api.queries.WallObjectQuery;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.util.QueryRunner;
+
+import javax.inject.Inject;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.util.List;
+
+class BarrowsOverlay extends Overlay
+{
+	private static final int MAX_DISTANCE = 2350;
+
+	private final Client client;
+	private final BarrowsPlugin plugin;
+	private final BarrowsConfig config;
+
+	@Inject
+	QueryRunner queryRunner;
+
+	@Inject
+	BarrowsOverlay(Client client, BarrowsPlugin plugin, BarrowsConfig config)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		this.client = client;
+		this.plugin = plugin;
+		this.config = config;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics, Point parent)
+	{
+		if (!config.enabled())
+		{
+			return null;
+		}
+
+		Player local = client.getLocalPlayer();
+
+		//Determine if you are in the tunnels based off if a barrows wall is present
+		if (inTunnels(local) && config.showMinimap())
+		{
+			//NPC yellow dot
+			List<NPC> npcs = client.getNpcs();
+			for (NPC npc : npcs)
+			{
+				net.runelite.api.Point minimapLocation = npc.getMinimapLocation();
+				if (minimapLocation != null)
+				{
+					graphics.setColor(Color.yellow);
+					graphics.fillOval(minimapLocation.getX(), minimapLocation.getY(), 4, 4);
+				}
+			}
+
+			//Render barrows walls/doors
+			renderObjects(graphics, local);
+
+			//Player white dot
+			graphics.setColor(Color.white);
+			graphics.fillRect(local.getMinimapLocation().getX(), local.getMinimapLocation().getY(), 3, 3);
+		}
+		else if (config.showBrotherLoc())
```

Comment:
This part needs to be split into separate overlay with layer ABOVE_SCENE

Reply:
Both of these are overtop of the minimap, so ABOVE_WIDGETS is the correct layer